extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeImage.php

```diff
@@ -12,41 +12,118 @@ class NodeImage extends Node {
 
 	public function getData() {
 		if ( !isset( $this->data ) ) {
-			$imageData = $this->getRawValueWithDefault( $this->xmlNode );
-
-			if( is_string($imageData) && PortableInfoboxDataBag::getInstance()->getGallery($imageData)) {
-				$imageData = PortableInfoboxDataBag::getInstance()->getGallery($imageData);
+			$this->data = array();
+
+			$parameterValue = $this->getRawValueWithDefault( $this->xmlNode );
+
+			if ( $this->containsTabberOrGallery( $parameterValue ) ) {
+				$parsed = $this->getExternalParser()->parseRecursive( $parameterValue );
+
+				$items = $this->getGalleryItems( $parsed ) + $this->getTabberItems( $parsed );
+				for( $i = 0; $i < count( $items ); $i++ ) {
+					$this->data[] = $this->getImage(
+						$items[$i]['title'],
+						$items[$i]['caption'],
+						$items[$i]['caption']
+					);
+				}
+			} else {
+				$this->data[] = $this->getImage(
+					$parameterValue,
+					$this->getValueWithDefault( $this->xmlNode->{self::ALT_TAG_NAME} ),
+					$this->getValueWithDefault( $this->xmlNode->{self::CAPTION_TAG_NAME} )
+				);
 			}
+		}
+
+		return $this->data;
+	}
+
+	private function getGalleryItems( $html ) {
```

Comment:
Can we plug into the gallery earlier than at the html stage, per discussion in https://github.com/Wikia/app/pull/8595 ?